src/DeviceRendererFactory.js

```diff
@@ -210,39 +190,36 @@ module.exports = class DeviceRendererFactory {
      * @param  {DeviceRenderer}     instance The DeviceRenderer instance reference to link into each plugin.
      * @param  {Object}             options  Various configuration options.
      */
-    addPlugins(instance, options) {
+    addPlugins(instance) {
         /*
          * Load instance dedicated plugins
          */
 
         const pluginInitMap = [
-            {enabled: options.touch, class: MultiTouchEvents},
-            {enabled: options.fullscreen, class: Fullscreen},
-            {enabled: options.clipboard, class: Clipboard, params: [options.i18n]},
-            {enabled: options.fileUpload, class: FileUpload, params: [options.i18n]},
-            {enabled: options.battery, class: Battery, params: [options.i18n]},
-            {enabled: options.streamBitrate, class: StreamBitrate, params: [options.i18n]},
-            {enabled: options.gps, class: GPS, params: [options.i18n, options.gpsSpeedSupport]},
-            {enabled: options.capture, class: Screencast, params: [options.i18n]},
-            {enabled: options.identifiers, class: Identifiers, params: [options.i18n]},
-            {enabled: options.network, class: Network, params: [options.i18n]},
-            {enabled: options.phone, class: Phone, params: [options.i18n]},
-            {enabled: options.baseband, class: BasebandRIL, params: [options.i18n, options.baseband]},
-            {enabled: options.streamResolution, class: StreamResolution},
-            {enabled: options.diskIO, class: IOThrottling, params: [options.i18n]},
-            {enabled: options.gamepad, class: GamepadManager},
-            {enabled: options.biometrics, class: FingerPrint},
-            {enabled: options.buttons, class: ButtonsEvents, params: [options.i18n, options.translateHomeKey]},
-            {enabled: options.camera || options.microphone, class: MediaManager},
         ];
 
         if (typeof instance.addCustomPlugins === 'function') {
-            instance.addCustomPlugins();
+            const customPlugins = instance.addCustomPlugins();
+            if (Array.isArray(customPlugins)){
```

Comment:
Maybe `customPlugins` could be simplified to an array, instead of having an `addCustomPlugins` method that generates an array?

Reply:
it's to avoid to break actual implementation, cause actually addCustomPlugins load plugins

Reply:
Ok. I don't actually mind if we break the current implementation, since I'll be working on the Shadow player very soon and could update it next week. And in the meantime it wouldn't break stuff since the player version is pinned.

Reply:
As discussed, i prefere that addCustomPlugins stay a function. It would be cleaner if we need to put logic before return the array. Plus it could be use as a hook, but it's more a trick